### PR TITLE
[Merged by Bors] - chore: avoid some `(0 < ((↑) : ℝ≥0 → ℝ) x) = (0 < x)` defeq abuses

### PR DESCRIPTION
**Pull request by astrainfinita**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Data/NNReal/Basic.lean [hunk 1]
```diff
@@ -336,6 +336,8 @@ noncomputable example : LinearOrder ℝ≥0 := by infer_instance
 
 @[bound] private alias ⟨_, Bound.coe_pos_of_pos⟩ := coe_pos
 
+lemma coe_pos_iff_ne_zero : (0 : ℝ) < r ↔ r ≠ 0 := coe_pos.trans pos_iff_ne_zero
```
Comment (YaelDillies) [inline Mathlib/Data/NNReal/Basic.lean:339]: Please don't mix the `<` and `≠` spellings. Either prove `(0 : ℝ) < r ↔ 0 < r` or `r ≠ (0 : ℝ) ↔ r ≠ 0`, but not something in between

**Comment by YaelDillies** [comment]
I must say the new proofs seem a bit pedantic and I'm not sure this PR is worth merging, but 🤷

**Pull request closed without merging**